iOS/LocalPackages/AIChat/Sources/AIChat/AIChatWebViewController.swift

```diff
@@ -57,10 +60,26 @@ final class AIChatWebViewController: UIViewController {
         super.viewDidLoad()
 
         view.backgroundColor = .black
-
         setupWebView()
         setupLoadingView()
         loadWebsite()
+        setupDownloadHandler()
+    }
+
+    private func setupDownloadHandler() {
+        downloadHandler.onDownloadComplete = { [weak self] result in
+            guard let self = self else { return }
+
+            switch result {
+            case .success(let filename):
+                self.view.showDownloadCompletionToast(for: filename) {
+                    self.delegate?.aiChatWebViewController(self, didFinishDownload: filename)
```

Comment:
I find the naming of the delegates a bit misleading here:
1. This line is actually an `onButtonTapped` closure of the toast call
2. It means that the `self.delegate?.aiChatWebViewController(self, didFinishDownload: filename)` is not being called when the download has finished but when user tapped "Show" on the toast after the download has finished.

My suggestion would be to rename delegate methods (all in the chain) to denote that it is a request to open downloads.

Reply:
Good point, doing the change now